### PR TITLE
FOLSPRINGB-127 - Update existing system-user credentials based on provided system user environment variable.

### PR DESCRIPTION
**Pull request by gurleenkaurbp**
## Purpose
_Describe the purpose of the pull request. Include background information if necessary._
The purpose of this PR is to make code changes as per the need of FOLSPRINGB-127.
## Approach
_How does this change fulfill the purpose?_
If the user is existing and the system user password has been specified in the environment variables then, the existing users credentials should be deleted and the credentials should be recreated as per the username, password specified in environment variables.
#### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

## Learning
_Describe the research stage. Add links to blog posts, patterns, libraries or addons used to solve this problem._

### CODE REVIEW

File: folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java [hunk 1]
```diff
@@ -114,6 +117,25 @@ void shouldAddOnlyNewPermissions() {
...
+    when(usersClient.query(any())).thenReturn(userExistsResponse());
+    when(permissionsClient.getUserPermissions(any()))
+      .thenReturn(asSinglePage("inventory-storage.instance.item.get"));
+    prepareSystemUser(systemUserProperties());
+    verify(authnClient).deleteCredentials(any());
+    verify(authnClient).saveCredentials(any());
+  }
+
+  @Test
+  void donNotupdateCredentialsForAnExistingUserWithoutSystemUserPassword() {
```
Comment (steveellis) [inline folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java:131]: Typo. This method should start as `doNotUpdate`.

Comment (steveellis) [inline folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java:131]:
```suggestion
  void doNotUpdateCredentialsForAnExistingUserWithoutSystemUserPassword() {
```

File: folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java [hunk 2]
```diff
@@ -114,6 +117,25 @@ void shouldAddOnlyNewPermissions() {
...
+    when(usersClient.query(any())).thenReturn(userExistsResponse());
+    when(permissionsClient.getUserPermissions(any()))
+      .thenReturn(asSinglePage("inventory-storage.instance.item.get"));
+    prepareSystemUser(systemUserPropertiesWithoutPassword());
+    verifyNoInteractions(authnClient);
```
Comment (steveellis) [inline folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java:136]:
```suggestion
      verify(authnClient, never()).deleteCredentials(any());
      verify(authnClient, never()).saveCredentials(any());
```
Might be more symmetrical with the positive test case.

File: folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java (hunk 1 above)
Comment (gurleenkaurbp) [reply folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java]: done

File: folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java (hunk 2 above)
Comment (gurleenkaurbp) [reply folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java]: Done

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

File: folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java [hunk 3]
```diff
@@ -114,6 +118,26 @@ void shouldAddOnlyNewPermissions() {
...
+    when(usersClient.query(any())).thenReturn(userExistsResponse());
+    when(permissionsClient.getUserPermissions(any()))
+      .thenReturn(asSinglePage("inventory-storage.instance.item.get"));
+    prepareSystemUser(systemUserProperties());
+    verify(authnClient).deleteCredentials(any());
+    verify(authnClient).saveCredentials(any());
+  }
+
+  @Test
+  void donotUpdateCredentialsForAnExistingUserWithoutSystemUserPassword() {
```
Comment (steveellis) [inline folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java:132]:
```suggestion
  void doNotUpdateCredentialsForAnExistingUserWithoutSystemUserPassword() {
```
No big deal. Just noticing

**Review by steveellis — APPROVED** [review]

File: folio-spring-system-user/README.md [hunk 4]
```diff
@@ -27,6 +27,7 @@ Requirements:
     * users.collection.get
     * users.item.post
     * login.item.post
+    * login.item.delete
```
Comment (psmagin) [inline folio-spring-system-user/README.md:30]: @gurleenkaurbp, how it's expected to be updated in modules? Will you create tickets for modules that use this functionality?

**Review by psmagin — APPROVED** [review]

File: folio-spring-system-user/README.md (hunk 4 above)
Comment (gurleenkaurbp) [reply folio-spring-system-user/README.md:30]: @psmagin , 
I am not creating any tickets at the moment for the modules. While my team was working on MODINREACH-403 we came across this scenario and hence decided to fix it. For the other modules if existing system username is being used and the credentials will be tried to be updated then there can be chance they can see permission related issue if this permission is not present in its respective module. 
@steveellis - What do you propose to do will it be an announcement that this "login.item.delete" permission added in the library code and other modules using this library could add the permission if its missing ?

**Pull request merged**